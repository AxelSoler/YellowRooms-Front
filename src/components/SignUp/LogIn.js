import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { IconContext } from 'react-icons';
import { ImCross } from 'react-icons/im';
import { loginUser } from '../../Redux/Authenticate/authentication';
import '../Splash/Splash.css';

const LogIn = (props) => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const initialForm = {
    user: {
      login: '',
      password: '',
    },
  };
  const [user, setUser] = useState(initialForm);
  const handleUserChange = (e) => {
    setUser((old) => ({
      ...old,
      user: { ...old.user, [e.target.name]: e.target.value },
    }));
  };
  // NOTE: Navigate to rooms on success log in
  const changeNavigation = () => {
    navigate('/rooms');
  };
  const userSubmit = async (e) => {
    e.preventDefault();
    dispatch(loginUser(user, changeNavigation));
  };
  const { style } = props;

  return (
    <div className={style}>
      <form className="login-form" onSubmit={userSubmit}>
        <div className="signupTitle">
          <h3 className="signup">LogIn</h3>
          <IconContext.Provider value={{ size: '24px', className: 'crossBtn' }}>
            <ImCross />
          </IconContext.Provider>
        </div>
        <input
          onChange={handleUserChange}
          className="loginInput"
          name="login"
          value={user.user.login}
          type="text"
          placeholder="User"
          required
        />
        <input
          onChange={handleUserChange}
          className="loginInput"
          type="password"
          name="password"
          value={user.user.password}
          placeholder="Password"
          required
        />
        <button className="registerBtn" type="submit">LogIn</button>
      </form>
    </div>
  );
};

export default LogIn;

LogIn.propTypes = {
  style: PropTypes.string,
};

LogIn.defaultProps = {
  style: 'inactive',
};
