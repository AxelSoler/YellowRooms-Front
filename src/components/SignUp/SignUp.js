import React, { useState, useEffect } from "react";
import { useNavigate } from "react-router-dom";
import { registerUser } from "../../Redux/Authenticate/authentication";
import { useDispatch, useSelector } from "react-redux";
const SignUp = () => {
  const dispatch = useDispatch();
  const currentUser = useSelector((state) => state.user);
  const navigate = useNavigate();
  const [user, setUser] = useState({
    user: {
      username: "",
      email: "",
      password: "",
    },
  });
  const handleUserChange = (e) => {
    setUser((old) => ({
      ...old,
      user: { ...old.user, [e.target.name]: e.target.value },
    }));
  };

  const changeNavigation = () => {
    navigate("/rooms");
  };
  const userSubmit = async (e) => {
    e.preventDefault();
    dispatch(registerUser(user, changeNavigation));
  };

  return (
    <div className="signup-page">
      <div>SignUp</div>
      <form onSubmit={userSubmit}>
        <input
          name="username"
          id="username"
          onChange={handleUserChange}
          className="signInput"
          type="text"
          placeholder="User"
          required
        />
        <input
          name="email"
          id="email"
          onChange={handleUserChange}
          className="signInput"
          type="email"
          placeholder="Email"
          required
        />
        <input
          name="password"
          id="password"
          onChange={handleUserChange}
          className="signInput"
          type="password"
          placeholder="Password"
          required
        />
        <button type="submit">Register</button>
      </form>
    </div>
  );
};

export default SignUp;
