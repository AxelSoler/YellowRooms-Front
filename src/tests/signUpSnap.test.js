import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import SignUp from '../components/SignUp/SignUp';

test('SignUp snapshot', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <SignUp />
        ,
      </MemoryRouter>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
