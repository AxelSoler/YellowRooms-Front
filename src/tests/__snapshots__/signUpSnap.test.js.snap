// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp snapshot 1`] = `
Array [
  <div
    className="signup-page"
  >
    <div>
      SignUp
    </div>
    <form
      onSubmit={[Function]}
    >
      <input
        className="signInput"
        id="username"
        name="username"
        onChange={[Function]}
        placeholder="User"
        required={true}
        type="text"
      />
      <input
        className="signInput"
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="email"
      />
      <input
        className="signInput"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
      />
      <button
        type="submit"
      >
        Register
      </button>
    </form>
  </div>,
  ";",
]
`;
